### PR TITLE
Add argument to specify tenant

### PR DESCRIPTION
**Pull request by stuartleeks**
Add argument to specify which tenant to load subscriptions for and add/update docs

### CODE REVIEW

**Comment by lawrencegripper** [comment]
I WANT THIS NOW. LET ME TEST

File: pkg/armclient/armclient.go [hunk 1]
```diff
@@ -33,12 +33,15 @@ type Client struct {
 
 // LegacyInstance is a singleton ARMClient used while migrating to the
 // injected client
-var LegacyInstance = NewClientFromCLI()
+var LegacyInstance Client
 
 // NewClientFromCLI creates a new client using the auth details on disk used by the azurecli
-func NewClientFromCLI() *Client {
+func NewClientFromCLI(subscription string) *Client {
```
Comment (lawrencegripper) [inline pkg/armclient/armclient.go]: Nit: this looks like it takes in a `tenantID` so param should this be `tenantID` or have I misunderstood

Comment (stuartleeks) [reply pkg/armclient/armclient.go]: Darn - hangover from initial spike on this!

**Comment by stuartleeks** [comment]
Fixed up the misnamed parameter and rebased on latest master :)

**Pull request merged**